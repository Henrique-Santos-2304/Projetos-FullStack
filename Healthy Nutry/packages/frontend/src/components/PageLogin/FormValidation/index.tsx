import * as S from "./styles";
import { useRef, useState } from "react";

import InputsLogin from "components/globalComponents/InputsLogin";

import { useForm } from "react-hook-form";
import { yupResolver } from "@hookform/resolvers/yup";
import * as Yup from "yup";
import { useUserContext } from "hooks/userContext";
import { requestGetUser, requestPostResult } from "api/requestsApi";

const schema = Yup.object({
  user: Yup.string()
    .required("Por favor digite um nome de usuário")
    .min(3, "O nome deve ter mais de 3 letras"),
  password: Yup.string()
    .required("Por favor digite uma senha")
    .min(6, "A senha deve ter pelo menos 6 caracteres"),
}).required();

type FormData = {
  user: string;
  password: string;
};

const FormValidation = () => {
  const { verifyDataEquals } = useUserContext();
  const [errorUserName, setErrorUserName] = useState<string>("");
  const [errorPassword, setErrorPassword] = useState<string>("");

  const {
    handleSubmit,
    register,
    formState: { errors },
  } = useForm<FormData>({ resolver: yupResolver(schema) });
  const formRef = useRef<HTMLFormElement>(null);

  const getUserBd = async ({ user, password }: FormData) => {
    try {
      const data: requestPostResult = await requestGetUser(user, password);

      data.errorMessage && setErrorPassword(data.errorMessage);
      data.user && verifyDataEquals(data.user);
    } catch (e) {
      setErrorUserName("Erro ao buscar usúario");
      console.log(e);
    }
  };

  const onSubmit = handleSubmit((user) => getUserBd(user));

  return (
    <S.Form onSubmit={onSubmit} ref={formRef}>
      <InputsLogin error={errors.user && errors.user.message}>
        <S.Inputs
          id="user"
          type="text"
          placeholder="Digite seu nome de usuário"
          {...register("user", {
            required: "Por favor digite seu nome de usuário",
          })}
        />
      </InputsLogin>
      {errorUserName && <S.Error>{errorUserName}</S.Error>}
      {}

      <div style={{ margin: "3rem 0" }} />

      <InputsLogin error={errors.password && errors.password.message}>
        <S.Inputs
          id="password"
          type="password"
          placeholder="Digite sua senha"
          {...register("password", {
            required: "Por favor digite sua senha",
          })}
        />
      </InputsLogin>
      {errorPassword && <S.Error>{errorPassword}</S.Error>}
      <S.Button type="submit" value="Enviar" />
    </S.Form>
  );
};

export default FormValidation;
