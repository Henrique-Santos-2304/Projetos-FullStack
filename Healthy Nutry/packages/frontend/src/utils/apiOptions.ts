export const optionsMealTypes = [
  { value: "", label: "Selecione tipo de refeição" },
  { value: "coffee", label: "Café" },
  { value: "lunch", label: "Almoço" },
  { value: "dinner", label: "Janta" },
];

export const optionsProteins = [
  { value: "", label: "Selecione uma Proteína" },
  { value: "fish", label: "Peixe" },
  { value: "ox", label: "Boi" },
  { value: "pig", label: "Porco" },
  { value: "chicken", label: "Frango" },
  { value: "egg", label: "Ovo" },
];

export const optionsCarbohydrates = [
  { value: "", label: "Selecione um Carboidrato" },
  { value: "rice", label: "Arroz" },
  { value: "potato", label: "Batata" },
  { value: "Yam", label: "Inhame" },
  { value: "noodle", label: "Macarrão" },
  { value: "bread", label: "Pão" },
];

export const optionsVegetables = [
  { value: "", label: "Selecione um Vegetal" },
  { value: "zucchini", label: "Abobrinha" },
  { value: "aubergine", label: "Berinjela" },
  { value: "pumpkin", label: "Abóbora" },
  { value: "chicory", label: "Almeirão" },
  { value: "tomato", label: "Tomate" },
];

export const proteinsChoice = ["Peixe", "Boi", "Porco", "Frango", "Ovo"];
export const carbohydrateChoice = [
  "Arroz",
  "Batata",
  "Inhame",
  "Macarrão",
  "Pão",
];
export const vegetableChoice = [
  "Abobrinha",
  "Berinjela",
  "Abóbora",
  "Almeirão",
  "Tomate",
];
