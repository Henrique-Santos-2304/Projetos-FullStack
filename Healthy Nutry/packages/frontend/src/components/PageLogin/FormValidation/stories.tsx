import { Story, Meta } from "@storybook/react";
import FormValidation from ".";

export default {
  title: "FormValidation",
  component: FormValidation,
} as Meta;

export const Basic: Story = (args) => <FormValidation {...args} />;

Basic.args = {
  title: "title basic",
  description: "description basic",
};

export const Default: Story = (args) => <FormValidation {...args} />;
