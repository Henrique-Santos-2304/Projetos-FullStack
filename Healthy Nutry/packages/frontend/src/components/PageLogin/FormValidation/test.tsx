import FormValidation from ".";
import { screen } from "@testing-library/react";
import { renderTheme } from "utils/testRenderTheme";

describe("<FormValidation />", () => {
  it("should render ", () => {
    const { container } = renderTheme(<FormValidation />);
    expect(container).toMatchSnapshot();
    screen.logTestingPlaygroundURL();
  });
});
