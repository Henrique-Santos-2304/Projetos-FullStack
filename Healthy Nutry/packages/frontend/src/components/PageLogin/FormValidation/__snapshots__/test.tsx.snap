// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormValidation /> should render  1`] = `
.c1 {
  padding: 0.8rem 1.6rem;
  border-bottom: 0.1rem solid #f8f8f8;
  font-size: 1.4rem;
  background: transparent;
  border-radius: 0.7rem;
  text-align: start;
  color: #f8f8f8;
  -webkit-transition: all 0.3s linear;
  transition: all 0.3s linear;
}

.c1::-webkit-input-placeholder {
  color: rgba(155,155,155,0.95);
}

.c1::-moz-placeholder {
  color: rgba(155,155,155,0.95);
}

.c1:-ms-input-placeholder {
  color: rgba(155,155,155,0.95);
}

.c1::placeholder {
  color: rgba(155,155,155,0.95);
}

.c1:hover,
.c1:focus {
  background: rgba(255,255,255,0.1);
  border: 0.1rem solid #b8f706;
}

.c2 {
  margin-bottom: 0.3rem;
  padding: 0.5rem 1.6rem;
  background: transparent;
  color: #b8f706;
  font-weight: 700;
  border-radius: 0.5rem;
  border: 0.1rem solid #b8f706;
  -webkit-transition: all 0.4s ease-in-out;
  transition: all 0.4s ease-in-out;
}

.c2:hover {
  color: #f8f8f8;
  border: 0.1rem solid #f8f8f8;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 1rem 0.6rem;
  margin: 0.6rem 0;
}

@media (min-width:620px) {
  .c1 {
    width: 24rem;
  }
}

@media (min-width:768px) {
  .c1 {
    font-size: 1.4rem;
  }
}

<div>
  <form
    class=""
  >
    <div
      class="c0"
    >
      <input
        class="c1"
        id="user"
        name="user"
        placeholder="Digite seu nome de usuÃ¡rio"
        type="text"
      />
    </div>
    <div
      class="c0"
    >
      <input
        class="c1"
        id="password"
        name="password"
        placeholder="Digite sua senha"
        type="password"
      />
    </div>
    <input
      class="c2"
      type="submit"
      value="Enviar"
    />
  </form>
</div>
`;
